bundle/bundle.go

```diff
@@ -343,6 +413,38 @@ func (r *Reader) Read() (Bundle, error) {
 	return bundle, nil
 }
 
+func (r *Reader) checkSignaturesAndDescriptors(signatures *SignaturesConfig, descriptors []*Descriptor) error {
+	if r.skipVerify {
+		return nil
+	}
+
+	if signatures == nil && r.verificationConfig != nil {
+		return fmt.Errorf("bundle reader provided with signature verification config but .signatures.json file not included in the bundle")
+	}
+
+	if signatures != nil {
+		if r.verificationConfig == nil {
+			return fmt.Errorf("reader missing signature verification config (hint: check opa configuration includes key(s))")
```

Comment:
Let's make this error message consistent w/ the one returned by the loader; if we made it simply `verification key not provided` then the output from `opa run -b .` would be:

```
error: load error: bundle .:  verification key not provided
```

Another reason I prefer this approach for the errors is that currently we're referring to command-line arguments and configuration somewhat interchangeably (e.g., the loader references specific command-line flags whereas the bundle package refers to configuration). To some extent what we want are structured errors returned by the loader and bundle packages and then a layer in the CLI or presentation that can convert those structured errors into nice human-readable messages. That's obviously more work though so the right middle ground (IMO) is short, explicit messages (which maybe a bit generic but hopefully get to the point.)

Reply:
Updated